common/src/stack/command/stack/commands/load/json/plugin_global.py

```diff
@@ -0,0 +1,164 @@
+			if scope == 'attrs':
+				for attr in import_data[scope]:
+					attr_type = attr['type']
+
+					if attr_type == 'shadow':
+						attr_shadow = True
+					else:
+						attr_shadow = False
+
+					try:
+						self.owner.command('set.attr', [
+								f'attr={attr["attr"]}',
+								f'value={attr["value"]}',
+								f'shadow={attr_shadow}'
+						])
+						self.owner.log.info(f'success setting global attr {attr["attr"]}')
+						self.owner.successes += 1
+
+					except CommandError as e:
+						if 'exists' in str(e):
+							self.owner.log.info(f'warning setting global attr {attr["attr"]}: {e}')
+							self.owner.warnings += 1
+						else:
+							self.owner.log.info(f'error setting global attr {attr["attr"]}: {e}')
+							self.owner.errors += 1
+
+
+			elif scope == 'route':
+				for route in import_data[scope]:
+					try:
+						self.owner.command('add.route', [
+								f'address={route["network"]}',
+								f'gateway={route["gateway"]}',
+								f'netmask={route["netmask"]}'
+						])
+						self.owner.log.info(f'success adding global route {route["network"]}')
+						self.owner.successes += 1
+
+					except CommandError as e:
+						if 'exists' in str(e):
+							self.owner.log.info(f'warning adding global route {route["network"]}: {e}')
+							self.owner.warnings += 1
+						else:
+							self.owner.log.info(f'error adding global route {route["network"]}: {e}')
+							self.owner.errors += 1
+
+			elif scope == 'firewall':
+				for rule in import_data[scope]:
+					try:
+						parameters = [
+						f'action={rule["action"]}',
+						f'chain={rule["chain"]}',
+						f'protocol={rule["protocol"]}',
+						f'service={rule["service"]}',
+						f'network={rule["network"]}',
+						f'output-network={rule["output-network"]}',
+						f'rulename={rule["name"]}',
+						f'table={rule["table"]}'
+					]
+						self.owner.command('add.firewall', parameters)
+						self.owner.log.info(f'success adding global firewall fule {rule["name"]}')
+						self.owner.successes += 1
+
+					except CommandError as e:
+						if 'exists' in str(e):
+							# the firewall rule exists but we want to replace it
+							try:
+								self.owner.command('remove.firewall', [ f'rulename={rule["name"]}' ])
+								self.owner.command('add.firewall', parameters)
+								self.owner.log.info(f'success replacing global firewall rule {rule["name"]}')
+								self.owner.successes += 1
+							except CommandError as e:
+								self.owner.log.info(f'error adding global firewall rule {rule["name"]}: {e}')
+								self.owner.errors += 1
+						else:
+							self.owner.log.info(f'error adding global firewall rule {rule["name"]}: {e}')
+							self.owner.errors += 1
+
+			elif scope == 'partition':
+				for partition in import_data[scope]:
+					try:
+						# normally the scope would be the first argument but since we are in the global plugin we need to leave it blank. Stacki defaults to global
+						self.owner.command('add.storage.partition', [
```

Comment:
In the appliance load plugin, you check to see if some of these fields are defined.  Are they optional?  If so, you should do the same thing in each place you add partition info.